### PR TITLE
feat: V0.1.26 fixes

### PR DESCRIPTION
**Pull request by ncrispino**
# PR: v0.1.26 Fixes and Enhancements

## Summary

This PR includes several bug fixes and enhancements following the v0.1.26 release:

- **Session Sharing via GitHub Gist** - Share MassGen sessions with a single command (MAS-16)
- Add `massgen logs` command for viewing and analyzing run logs
- Add per-LLM call time tracking for detailed performance analysis
- Improve config building via CLI with better wizard flow
- Add context path support in the web UI wizard
- Fix Claude Code tool allow issue and add "Open in Browser" button
- Add Gemini 3 Flash support (MAS-168)
- Fix workflow error timeout handling (MAS-170)

Closes MAS-16, MAS-168, MAS-170

#### Session Sharing via GitHub Gist

Share MassGen session logs with collaborators via GitHub Gist. Sessions are viewable in the MassGen Viewer web app without requiring GitHub login.

```bash
# Share the most recent session
massgen export

# Share a specific session
massgen export log_20251218_134125_867383

# List your shared sessions
massgen shares list

# Delete a shared session
massgen shares delete <gist_id>
```

**Features:**
- Uploads session logs to GitHub Gist (requires `gh` CLI authenticated)
- Returns a shareable URL to the MassGen Viewer (`https://massgen.github.io/MassGen-Viewer/?gist=...`)
- Anyone with the link can view the session (no login required)
- Automatically excludes large files, debug logs, and sensitive data (API keys redacted)
- Includes workspace files with common extensions (.py, .js, .txt, etc.)
- Priority inclusion of key files: metrics, status, coordination events

**Files changed:**
- `massgen/share.py` - New module for GitHub Gist sharing (373 lines)
- `massgen/session_exporter.py` - Session export logic
- `massgen/cli.py` - Added `export` and `shares` commands
- `massgen/filesystem_manager/_constants.py` - Sharing constants and limits
- `docs/source/user_guide/logging.rst` - User documentation
- `docs/source/reference/cli.rst` - CLI reference documentation
- `docs/source/quickstart/running-massgen.rst` - Quickstart guide

#### `massgen logs` Command

A new CLI command for viewing and analyzing MassGen run logs with rich formatting:
- List recent logs with `massgen logs list`
- View specific log details with `massgen logs view <log_id>`
- Filter logs by status, date range, and search terms
- Export logs in JSON or CSV format
- Added documentation in `docs/source/user_guide/logging.rst`

**Files changed:**
- `massgen/cli.py` - Added logs command group
- `massgen/logs_analyzer.py` - New module for log analysis (433 lines)
- `docs/source/user_guide/logging.rst` - User documentation

#### Per-LLM Call Time Tracking

Added detailed timing metrics for individual LLM calls within a session:
- Track time spent on each LLM API call
- Aggregate timing statistics in metrics summary
- Helps identify performance bottlenecks across different models

**Files changed:**
- `massgen/backend/base.py` - Added timing instrumentation
- `massgen/backend/base_with_custom_tool_and_mcp.py` - Timing integration
- `massgen/backend/chat_completions.py` - OpenAI timing
- `massgen/backend/claude.py` - Anthropic timing
- `massgen/backend/gemini.py` - Google timing
- `massgen/backend/grok.py` - xAI timing
- `massgen/backend/response.py` - Response timing metadata
- `massgen/token_manager/token_manager.py` - Token manager timing aggregation
- `massgen/orchestrator.py` - Orchestrator timing collection
- `massgen/logs_analyzer.py` - Timing display in logs

#### Context Paths Support in Wizard

Added a new step in the quickstart wizard to configure context paths:
- New `ContextPathsStep` component for selecting workspace directories
- Enhanced wizard store to track context path configuration
- Improved file/workspace browser components

**Files changed:**
- `webui/src/components/wizard/ContextPathsStep.tsx` - New wizard step
- `webui/src/components/wizard/index.ts` - Export new component
- `webui/src/stores/wizardStore.ts` - Context paths state management
- `webui/src/components/QuickstartWizard.tsx` - Wizard flow updates
- `webui/src/components/FileWorkspaceBrowser.tsx` - Browser improvements

#### Gemini 3 Flash Support

Added support for Google's Gemini 3 Flash model:
- New provider config: `massgen/configs/providers/gemini/gemini_3_flash.yaml`
- Updated capabilities registry

**Files changed:**
- `massgen/backend/capabilities.py` - Added Gemini 3 Flash capabilities
- `massgen/configs/providers/gemini/gemini_3_flash.yaml` - New provider config

#### Enhanced Config Builder

Improved the CLI config building experience:
- Better interactive prompts and validation
- Enhanced logging configuration
- New skills installer utility for automated MCP skills setup

**Files changed:**
- `massgen/config_builder.py` - Significant enhancements (~234 lines added)
- `massgen/logger_config.py` - Improved logging setup
- `massgen/utils/skills_installer.py` - New utility for MCP skills installation
- `massgen/chat_agent.py` - Minor adjustments

#### "Open in Browser" Button

Added a button in the web UI to quickly open the browser tool:
- New browser launch endpoint in server
- Button in answer browser modals and final answer views

**Files changed:**
- `massgen/frontend/web/server.py` - New `/api/open-browser` endpoint
- `massgen/frontend/displays/web_display.py` - Display integration
- `webui/src/components/AnswerBrowserModal.tsx` - Added button
- `webui/src/components/FinalAnswerView.tsx` - Added button (also includes share functionality)
- `webui/src/components/InlineAnswerBrowser.tsx` - Added button

#### Claude Code Tool Allow Issue

Fixed an issue where Claude Code tool permissions weren't being properly handled:
- Added proper tool allow list handling in Claude Code backend
- Fixed filesystem manager edge cases

**Files changed:**
- `massgen/backend/claude_code.py` - Tool allow fix
- `massgen/backend/base.py` - Base backend adjustment
- `massgen/filesystem_manager/_filesystem_manager.py` - Edge case fixes
- `massgen/tests/test_backend_capabilities.py` - Updated tests

#### Workflow Error Timeout

Fixed timeout handling in the orchestrator for error respawn scenarios:
- Added proper timeout configuration for error recovery
- New debug config for testing error respawn behavior

**Files changed:**
- `massgen/orchestrator.py` - Timeout fix
- `massgen/configs/debug/error_respawn_test.yaml` - Debug test config

### Frontend Updates

- Rebuilt frontend assets with all UI changes
- Updated CSS and JavaScript bundles
- Added share button to FinalAnswerView component

**Files changed:**
- `massgen/frontend/web/static/assets/index-CfjyMu5p.js` - New bundle
- `massgen/frontend/web/static/assets/index-BhIeb3JU.css` - Updated styles
- `massgen/frontend/web/static/index.html` - Asset references

## Test Plan

- [ ] Test session sharing:
  - [ ] Run `massgen export` and verify gist is created
  - [ ] Open the share URL and verify session displays in MassGen Viewer
  - [ ] Run `massgen shares list` and verify shared sessions are listed
  - [ ] Run `massgen shares delete <gist_id>` and verify deletion
- [ ] Run `massgen logs list` and verify recent logs are displayed
- [ ] Run `massgen logs view <log_id>` and verify log details (including LLM timing) are shown
- [ ] Test config builder wizard with new context paths step
- [ ] Verify Gemini 3 Flash model works with a basic prompt
- [ ] Test "Open in Browser" button launches browser correctly
- [ ] Verify Claude Code tool permissions work as expected
- [ ] Run existing test suite: `uv run pytest massgen/tests/`

## Breaking Changes

None

## Migration Notes

None required. All changes are backwards compatible.

**Prerequisites for sharing:** The `gh` CLI must be installed and authenticated (`gh auth login`) to use the `massgen export` sharing feature.

### CODE REVIEW

File: docs/source/user_guide/logging.rst [hunk 1]
```diff
@@ -520,6 +616,128 @@ MassGen exits with status 0 on success, non-zero on failure.
...
+.. code-block:: bash
+
+   # Export to HTML file (no upload)
+   massgen export -o session.html
+
+   # Export and open in browser
+   massgen export -o session.html --open
+
+This creates a self-contained HTML file with all session data embedded.
+
```
Comment (ncrispino) [inline docs/source/user_guide/logging.rst]: is this still supported?

File: massgen/cli.py [hunk 2]
```diff
@@ -5057,6 +5144,9 @@ def cli_main():
         help="Enable rate limiting (uses limits from rate_limits.yaml config)",
     )
 
+    # Note: 'logs' subcommand is handled separately at the start of cli_main()
+    # to avoid conflict with the positional 'question' argument
+
```
Comment (ncrispino) [inline massgen/cli.py]: remove

File: massgen/configs/providers/gemini/gemini_3_flash.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      command_line_execution_mode: "docker"
+
+  # - id: "agent_b"
+  #   backend:
+  #     type: "gemini"
+  #     model: "gemini-3-flash-preview"
+  #     cwd: "workspace2"  # Required for filesystem mode
+  #     enable_mcp_command_line: true
+  #     command_line_execution_mode: "docker"
+
```
Comment (ncrispino) [inline massgen/configs/providers/gemini/gemini_3_flash.yaml:20]: comment out

File: massgen/frontend/displays/web_display.py [hunk 4]
```diff
@@ -330,6 +330,12 @@ def show_final_answer(
             selected_agent: The agent selected as winner
         """
         self._final_answer = answer
+
+        # Check if consensus_reached was already emitted by stream_final_answer_chunk
+        # IMPORTANT: Check BEFORE assigning self._selected_agent, otherwise the condition
+        # always fails since we'd be checking "selected_agent and not selected_agent"
```
Comment (ncrispino) [inline massgen/frontend/displays/web_display.py]: remove

File: massgen/share.py [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+    "execution_metadata.yaml",
+]
+
+# Files to exclude from agent_outputs (redundant)
+EXCLUDE_FILES = {
+    "final_presentation_agent_a_latest.txt",
+    "final_presentation_agent_b_latest.txt",
+    "final_presentation_agent_c_latest.txt",
+    "final_presentation_agent_d_latest.txt",
+}
```
Comment (ncrispino) [inline massgen/share.py]: dont match these just match the final presentation latest pattern
